src/dxtbx/model/scan_helpers.h

```diff
@@ -33,23 +33,81 @@ namespace dxtbx { namespace model {
    * Check if the angle is within the given range. The angular
    * range can be any two angles, plus or minus. The angle is to check can
    * also be any angle. The angle is considered within the range if the range
-   * spans more than 2PI degrees and the angle is within the two range angles
+   * spans more than 2PI degrees or the angle is within the two range angles
    * when mod 2PI.
    * @param range The angular range
    * @param angle The angle to check
    * @returns True/False the angle is within the range
    */
+
   inline bool is_angle_in_range(vec2<double> range, double angle) {
-    double diff_range0 = angle - range[0];
-    double diff_range1 = angle - range[1];
-    if (std::abs(diff_range0) < 2.0 * std::numeric_limits<double>::epsilon()) {
-      diff_range0 = 0.0;
+    auto is_number = [](double value, double number, double eps) {
```

Comment:
Can we call this `is_approx`/`approx_equal`/`approx`, to make it clearer what this does? Since `eps` is only ever a fixed value in this function, it could probably be also omitted...